google_guest_agent/network/manager/systemd_networkd_linux.go

```diff
@@ -169,27 +233,271 @@ func (n systemdNetworkd) IsManaging(ctx context.Context, iface string) (bool, er
+	}
+
+	configExp := `(?P<priority>[0-9]+)-(?P<interface>.*\.[0-9]+)-(?P<suffix>.*)\.(?P<extension>network|netdev)`
+	configRegex := regexp.MustCompile(configExp)
+
+	for _, file := range files {
+		var (
+			currIface, extension string
+			found                bool
+		)
+
+		if file.IsDir() {
+			continue
+		}
+
+		groups := utils.RegexGroupsMap(configRegex, file.Name())
+
+		// If we don't have a matching interface skip it.
+		if currIface, found = groups["interface"]; !found {
+			continue
+		}
+
+		// If we don't have a matching extension skip it.
+		if extension, found = groups["extension"]; !found {
+			continue
+		}
+
+		// If this is an interface still present skip it.
+		if slices.Contains(keepMe, currIface) {
+			continue
+		}
+
+		ptrMap := map[string]guestAgentManaged{
+			"network": new(systemdConfig),
+			"netdev":  new(systemdNetdevConfig),
+		}
+
+		ptr, foundPtr := ptrMap[extension]
+		if !foundPtr {
+			return fmt.Errorf("regex matching failed, invalid etension: %s", extension)
+		}
+
+		filePath := path.Join(n.configDir, file.Name())
+		if err := n.readIniFile(ptr, filePath); err != nil {
+			return fmt.Errorf("failed to read .network file before removal: %+v", err)
+		}
+
+		// Although the file name is following the same pattern we are assuming this is not
+		// managed by us - skip it.
+		if !ptr.isGuestAgentManaged() {
+			continue
+		}
+
+		if err := os.Remove(path.Join(n.configDir, file.Name())); err != nil {
+			return fmt.Errorf("failed to remove vlan interface config(%s): %+v", file.Name(), err)
+		}
+	}
+
+	return nil
+}
+
+// netdevFile returns the systemd's .netdev file path.
+// Priority is lexicographically sorted in ascending order by file name. So a configuration
+// starting with '1-' takes priority over a configuration file starting with '10-'. Setting
+// a priority of 1 allows the guest-agent to override any existing default configurations
+// while also allowing users the freedom of using priorities of '0...' to override the
+// agent's own configurations.
+func (n systemdNetworkd) netdevFile(iface string) string {
+	return path.Join(n.configDir, fmt.Sprintf("%d-%s-google-guest-agent.netdev", n.priority, iface))
+}
+
+// networkFile returns the systemd's .network file path.
+// Priority is lexicographically sorted in ascending order by file name. So a configuration
+// starting with '1-' takes priority over a configuration file starting with '10-'. Setting
+// a priority of 1 allows the guest-agent to override any existing default configurations
+// while also allowing users the freedom of using priorities of '0...' to override the
+// agent's own configurations.
+func (n systemdNetworkd) networkFile(iface string) string {
+	return path.Join(n.configDir, fmt.Sprintf("%d-%s-google-guest-agent.network", n.priority, iface))
+}
+
+// readIniFile reads and parses the content of filePath and loads it into ptr.
+func (n systemdNetworkd) readIniFile(ptr any, filePath string) error {
```

Comment:
`readIniFile` and `writeIniFile` could probably be pulled out into `common.go` and re-used in the `network-manager` setup.

Reply:
done. thanks.